<f:ajax> in composite component in <h:dataTable rowStatePreserved="true"> leads to InstantiationException during RENDER_RESPONSE

If a `<f:ajax>` tag is nested in a composite component that is placed in a `<h:dataTable>` with `rowStatePreserved="true"`, an `InstantiationException` is thrown during `RENDER_RESPONSE`.

# To reproduce

XHTML test case:
```
<h:dataTable
	rowStatePreserved="true"
	value="#{['1']}">
	<h:column>
		<my:compositecomponent>
			<f:ajax/>
		</my:compositecomponent>
	</h:column>
</h:dataTable>
```

Stack Trace:
```
java.lang.IllegalStateException: java.lang.InstantiationException: com.sun.faces.facelets.tag.composite.RetargetedAjaxBehavior
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.StateHolderSaver.restore(StateHolderSaver.java:129)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponentBase.restoreAttachedState(UIComponentBase.java:1425)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponentBase.restoreBehaviorsState(UIComponentBase.java:1752)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponentBase.restoreState(UIComponentBase.java:1224)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIData.restoreFullDescendantComponentStates(UIData.java:1468)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIData.restoreFullDescendantComponentStates(UIData.java:1477)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIData.restoreFullDescendantComponentStates(UIData.java:1477)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIData.restoreFullDescendantComponentStates(UIData.java:1477)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIData.restoreFullDescendantComponentStates(UIData.java:1477)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIData.setRowIndexRowStatePreserved(UIData.java:573)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIData.setRowIndex(UIData.java:475)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.renderkit.html_basic.TableRenderer.encodeChildren(TableRenderer.java:117)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:556)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:257)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:82)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:556)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:257)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:82)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:556)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1435)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.render.Renderer.encodeChildren(Renderer.java:146)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:556)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1435)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1438)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1438)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:449)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:160)
...
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:125)
        at jakarta.faces.impl@4.0.7.redhat-00001//com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:93)
...
Caused by: java.lang.InstantiationException: com.sun.faces.facelets.tag.composite.RetargetedAjaxBehavior
        at java.base/java.lang.Class.newInstance(Class.java:639)
        at jakarta.faces.impl@4.0.7.redhat-00001//jakarta.faces.component.StateHolderSaver.restore(StateHolderSaver.java:127)
        ... 109 more
Caused by: java.lang.NoSuchMethodException: com.sun.faces.facelets.tag.composite.RetargetedAjaxBehavior.<init>()
        at java.base/java.lang.Class.getConstructor0(Class.java:3585)
        at java.base/java.lang.Class.newInstance(Class.java:626)
        ... 110 more
```

The `StateHolderSaver` effectively demands from classes that implement `StateHolder` that they provide a default constructor. The exception arises when it tries to instantiate the class `RetargetedAjaxBehavior` via a default constructor (that does not exist).

# Context

Mojarra version: 4.0.7
Potentially caused by https://github.com/eclipse-ee4j/mojarra/pull/5294

--- a/impl/src/main/java/com/sun/faces/facelets/tag/composite/RetargetedAjaxBehavior.java
+++ b/impl/src/main/java/com/sun/faces/facelets/tag/composite/RetargetedAjaxBehavior.java
@@ -17,6 +17,7 @@
 package com.sun.faces.facelets.tag.composite;
 
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Set;
 
@@ -43,6 +44,11 @@ public class RetargetedAjaxBehavior extends AjaxBehavior {
 
     private AjaxBehavior retargeted;
     private List<String> targetClientIds;
+
+    public RetargetedAjaxBehavior() {
+        retargeted = new AjaxBehavior();
+        targetClientIds = Collections.emptyList();
+    }
 
     public RetargetedAjaxBehavior(AjaxBehavior retargeted, List<String> targetClientIds) {
         this.retargeted = retargeted;

--- a/impl/src/main/java/com/sun/faces/facelets/tag/composite/RetargetedAjaxBehavior.java
+++ b/impl/src/main/java/com/sun/faces/facelets/tag/composite/RetargetedAjaxBehavior.java
@@ -46,8 +46,7 @@ public class RetargetedAjaxBehavior extends AjaxBehavior {
     private List<String> targetClientIds;
 
     public RetargetedAjaxBehavior() {
-        retargeted = new AjaxBehavior();
-        targetClientIds = Collections.emptyList();
+        this(new AjaxBehavior(), Collections.emptyList());
     }
 
     public RetargetedAjaxBehavior(AjaxBehavior retargeted, List<String> targetClientIds) {

